plugins/cilium-cni/cilium-cni.go

```diff
@@ -488,18 +488,10 @@ func cmdAdd(args *skel.CmdArgs) (err error) {
 		res.Routes = append(res.Routes, routes...)
 	}
 
-	switch conf.IpamMode {
```

Comment:
I'm not sure if this should be removed. Don't think we need to run `eniAdd` when we're not running in `IpamMode`. `eniAdd` should not be called when Cilium is running in non native-routing mode. @tgraf can correct me here.

Reply:
I think you're perfectly right; gated back the `eniAdd` on aws|azure IpamMode.

Reply:
@bpineau did you give this a run in a real azure cluster? I'm interested how the `eniAdd` method behaves because I know it has some AWS `opinions` related to interface indexes.

Reply:
It's my understanding that `retrieveIfIndexFromMAC(mac, mtu)` finds the interface index from the mac address we provide, sets the mtu, then returns the numeric index, which is then used to name the routes rules/tables. Behaviour should be same on Azure. Index value mostly matters for mtu changes anyway -as long as the index remains stable-, since we can't really dynamically add interfaces on Azure.

Yes I run that code on a real Azure test cluster. This gives us the following, for instance:
```
# From a node running 2 (non hostnetwork) pods (10.194.0.176 and 10.194.0.167),
# eth1 in one of the pods subnets (10.194.0.0/20): pods can now join our other
# pods subnets (10.194.16.0/20, etc) attached to other instances, thanks to the
# new "default via 10.194.0.1 dev eth1" - same behaviour and routes as on AWS).

$ grep interface-name /etc/cni/net.d/10-generic-veth.conflist
        "interface-name": "eth1"

$ ip link list
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP mode DEFAULT group default qlen 1000
    link/ether 00:0d:3a:01:61:90 brd ff:ff:ff:ff:ff:ff
3: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP mode DEFAULT group default qlen 1000
    link/ether 00:0d:3a:01:6b:2d brd ff:ff:ff:ff:ff:ff

$ ip rule show
9:      from all fwmark 0x200/0xf00 lookup 2004
20:     from all to 10.194.0.176 lookup main
20:     from all to 10.194.0.167 lookup main
100:    from all lookup local
110:    from 10.194.0.176 lookup 3
110:    from 10.194.0.167 lookup 3
32766:  from all lookup main
32767:  from all lookup default

$ ip route show table 3
default via 10.194.0.1 dev eth1
10.194.0.1 dev eth1 scope link
```